packages/message-pickup-repository-dynamodb/src/client.ts

```diff
@@ -0,0 +1,208 @@
+    const startTime = Date.now()
+    const maxWaitTime = 30000
+
+    return new Promise((resolve, reject) => {
+      const checkTableStatus = async () => {
+        try {
+          const describeParams: DescribeTableCommandInput = {
+            TableName: this.tableName,
+          }
+          const command = new DescribeTableCommand(describeParams)
+          const response = await this.dynamodbClient.send(command)
+
+          if (response.Table?.TableStatus === 'ACTIVE') {
+            resolve()
+            return
+          }
+
+          if (Date.now() - startTime > maxWaitTime) {
+            reject(new Error(`Table ${this.tableName} did not become active within ${maxWaitTime}ms`))
+            return
+          }
+
+          setTimeout(checkTableStatus, 500)
+        } catch (error) {
+          reject(error)
+        }
+      }
+
+      checkTableStatus()
+    })
+  }
+
+  async getMessageCount(connectionId: string): Promise<number> {
+    const params: ScanCommandInput = {
+      TableName: this.tableName,
+      FilterExpression: 'connectionId = :connectionId',
+      ExpressionAttributeValues: {
+        ':connectionId': { S: connectionId },
+      },
+      Select: 'COUNT',
+    }
+
+    try {
+      const command = new ScanCommand(params)
+      const response = await this.dynamodbClient.send(command)
+      return response.Count || 0
+    } catch (error) {
+      console.error('Error getting entries count:', error)
+      throw error
+    }
+  }
+
+  async getMessages(options: {
+    connectionId: string
+    limit?: number
+    recipientDid?: string
+    deleteMessages?: boolean
+  }) {
+    const queryParams: QueryCommandInput = {
+      TableName: this.tableName,
+      KeyConditionExpression: 'connectionId = :connectionId',
+      ExpressionAttributeValues: {
+        ':connectionId': { S: options.connectionId },
+      },
+      Limit: options.limit,
+    }
+
+    if (options.recipientDid) {
+      queryParams.FilterExpression = 'contains(recipientDids, :recipientDid)'
+
+      if (queryParams.ExpressionAttributeValues) {
+        queryParams.ExpressionAttributeValues[':recipientDid'] = { S: options.recipientDid }
+      }
+    }
+
+    const command = new QueryCommand(queryParams)
+    const response = await this.dynamodbClient.send(command)
+
+    const messages = (response.Items?.map((item) => unmarshall(item)) || []).map(
+      (i) =>
+        ({
+          ...i,
+          receivedAt: new Date(i.receivedAt),
+          id: i.messageId.toString(),
+        }) as unknown as QueuedMessage
+    )
+
+    if (options.deleteMessages && messages.length > 0) {
+      await this.removeMessages({
+        connectionId: options.connectionId,
+        messageIds: messages.map((m) => m.id),
+      })
+    }
+
+    return messages
+  }
+
+  async addMessage(options: AddQueuedMessageOptions): Promise<string> {
+    const randomizer = randomInt(0, 255).toString().padStart(3, '0')
```

Comment:
where is this based on? Why 255 instead of 999? Is 256 options enough randomness

Reply:
Can do 999 as well. I think 1/255 is sufficient if they overlap in the ms, but 999 is even less.